### ISSUE TITLE
mautic:segments:update fails if a segments has an "Tags - including" filter with empty field value

### ISSUE THREAD

**Issue by alinmbb** [mautic/mautic#10951]
### Mautic Version

4.2.x series

### PHP version

7.4.27

### What browsers are you seeing the problem on?

Not relevant

### What happened?

We had some segments that under the Filters had a filter based on Tags with including condition. On the tags selection field it was empty. Mautic did not give any error/warning upon saving the segment. When the cronjob mautic:segments:update was executed, it gave an error :
 ```
 An exception occurred while executing 'SELECT count(leadIdPrimary) count, max(leadIdPrimary) maxId, min(leadIdPrimary) minId FROM (SELECT DISTINCT l.id  
   as leadIdPrimary, par5e.lead_id AS par5e_lead_id FROM leads l LEFT JOIN lead_lists_leads par5e ON par5e.lead_id = l.id and (par5e.leadlist_id = 240) W  
  HERE (EXISTS(SELECT NULL FROM lead_tags_xref par5d WHERE (par5d.lead_id = l.id) AND (par5d.tag_id IN ()))) AND (par5e.lead_id IS NULL)) sss':            
                                                                                                                                                           
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server   
  version for the right syntax to use near ')))) AND (par5e.lead_id IS NULL)) sss' at line 1      
```

As you can see, the resulted SQL contains   AND (par5d.tag_id IN ()))) AND . This is not valid SQL syntax, at least on our DB - 10.6.5-MariaDB.

This will intrrerrupt the cronjob execution and fail to do all the expected changes

### How can we reproduce this issue?

Step 1: Create a segment with an empty Tags including filter value. See picture, the name of the segment is not relevant.
![Screenshot 2022-03-10 at 13 26 00](https://user-images.githubusercontent.com/86683952/157664277-6742c579-dcb5-4677-b18f-00b6eeb3d937.png)
 Step 2: Run the mautic:segments:update cronjob.
Step 3: Notice the error

### Relevant log output

```shell
[2022-03-10 12:39:09] mautic.ERROR: Segment QB: Query Exception: An exception occurred while executing 'SELECT count(leadIdPrimary) count, max(leadIdPrimary) maxId, min(leadIdPrimary) minId FROM (SELECT DISTINCT l.id as leadIdPrimary, par5e.lead_id AS par5e_lead_id FROM leads l LEFT JOIN lead_lists_leads par5e ON par5e.lead_id = l.id and (par5e.leadlist_id = 240) WHERE (EXISTS(SELECT NULL FROM lead_tags_xref par5d WHERE (par5d.lead_id = l.id) AND (par5d.tag_id IN ()))) AND (par5e.lead_id IS NULL)) sss':  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')))) AND (par5e.lead_id IS NULL)) sss' at line 1 {"query":"SELECT count(leadIdPrimary) count, max(leadIdPrimary) maxId, min(leadIdPrimary) minId FROM (SELECT DISTINCT l.id as leadIdPrimary, par5e.lead_id AS par5e_lead_id FROM leads l LEFT JOIN lead_lists_leads par5e ON par5e.lead_id = l.id and (par5e.leadlist_id = 240) WHERE (EXISTS(SELECT NULL FROM lead_tags_xref par5d WHERE (par5d.lead_id = l.id) AND (par5d.tag_id IN ()))) AND (par5e.lead_id IS NULL)) sss","parameters":[]} []
[2022-03-10 12:39:09] mautic.NOTICE: Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing 'SELECT count(leadIdPrimary) count, max(leadIdPrimary) maxId, min(leadIdPrimary) minId FROM (SELECT DISTINCT l.id as leadIdPrimary, par5e.lead_id AS par5e_lead_id FROM leads l LEFT JOIN lead_lists_leads par5e ON par5e.lead_id = l.id and (par5e.leadlist_id = 240) WHERE (EXISTS(SELECT NULL FROM lead_tags_xref par5d WHERE (par5d.lead_id = l.id) AND (par5d.tag_id IN ()))) AND (par5e.lead_id IS NULL)) sss':  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ')))) AND (par5e.lead_id IS NULL)) sss' at line 1 (uncaught exception) at /var/www/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 98 while running console command `mautic:segments:update` [] []
[2022-03-10 12:39:09] mautic.WARNING: Command `mautic:segments:update` exited with status code 1 [] []
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by arthurocrates** [comment]
Same problem in contacts

<img width="1422" alt="Screen Shot 2022-03-10 at 14 15 31" src="https://user-images.githubusercontent.com/39348963/157738237-0d3f12ff-f03c-4637-b68e-5457457ffd9e.png">

**Comment by RCheesley** [comment]
@escopecz I think this was what I had observed when testing your segment refactoring PR. I think this is something we need to address as it's breaking people's segments which were already existing - sure they shouldn't have the blank field in the first place, but is there anything we can do to address the errors being thrown?

**Comment by escopecz** [comment]
Yes, this has to be addressed with a code change. The only way around it is to fix the segment filters in the user interface.

**Comment by gonbanchero** [comment]
hi, im having same problem here. 

Mautic Version: v4.2.2
Php Version: 7.4

Any ideas? dont know how to solve this problem.

<img width="1029" alt="Captura de Pantalla 2022-08-17 a la(s) 08 57 05" src="https://user-images.githubusercontent.com/85972008/185428819-bf34ad9a-5b7d-49a4-afa4-c0d22a53bbac.png">

<img width="1591" alt="Captura de Pantalla 2022-08-11 a la(s) 14 13 26" src="https://user-images.githubusercontent.com/85972008/185428804-c67e80c7-77d0-45be-ad9b-20a220d7811d.png">

thanks !

**Comment by RCheesley** [comment]
Pretty sure this was fixed in a later version to stop users making this mistake - but basically, don't use 'includes' without specifying something in the field. It should **not** be allowed to be created, but we missed the validation at some point. 

Clean up the segments that are throwing the errors, and remove the lines which are basically doing nothing (the ones with tags > includes and nothing is specified), and you won't get the error (and update to the latest Mautic!)

**Comment by gonbanchero** [comment]
hi Ruth, your help was really usefull.
i update, errase all wrong segments (no one have empty data in 'includes') but create them again and all is working great. 
thanks you a lot !

**Comment by Willianarkeit** [comment]
> Pretty sure this was fixed in a later version to stop users making this mistake - but basically, don't use 'includes' without specifying something in the field. It should **not** be allowed to be created, but we missed the validation at some point.
> 
> Clean up the segments that are throwing the errors, and remove the lines which are basically doing nothing (the ones with tags > includes and nothing is specified), and you won't get the error (and update to the latest Mautic!)



unfortunately not yet, I have the same problem and mautic 4.4.5 php 8.0

**Comment by dsp76** [comment]
same with 4.4.7

**Comment by escopecz** [comment]
This wasn't a regression, but will be fixed in Mautic 5.0.0
